django/urls/base.py

```diff
@@ -179,3 +183,50 @@ def translate_url(url, lang_code):
             else:
                 url = urlunsplit((parsed.scheme, parsed.netloc, url, parsed.query, parsed.fragment))
     return url
+
+
+_PATH_PARAMETER_COMPONENT_RE = re.compile(
+    '<(?:(?P<converter>[^:]+):)?(?P<parameter>[A-Za-z0-9_]+)>'
```

Comment:
Agreed. No reason to prevent non-ascii identifiers here. The extra check on `isidentifier` is a good one indeed.

For converters: I don't think there's a technical limitation that would force the converter "name" to actually be a valid identifier (or even limited to `\w+`), as long as it doesn't contain a colon. Alternatively, I expect actual usage to be limited to `[A-Za-z0-9_]` for both cases which seems to be a subset of `\w+`, so that should be ok.